### PR TITLE
Add missing documentation and description strings for all public classes

### LINKED ISSUE #173: Missing documentation for package operators and public classes
While reorganizing the project for packaging, I noticed the following operators and types remain undocumented:

- [`ContextTask`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/ContextTask.cs)
- [`CreateContext`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/CreateContext.cs)
- [`ConfigureHarpSyncInput`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/ConfigureHarpSyncInput.cs)
- [`ConfigureLoadTester`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/ConfigureLoadTester.cs)
- [`DeviceFactory`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/DeviceFactory.cs)
- [`HarpSyncInputData`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/HarpSyncInputData.cs)
- [`HarpSyncInputDataFrame`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/HarpSyncInputDataFrame.cs)
- [`HubDeviceFactory`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/HubDeviceFactory.cs)
- [`PassthroughState`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/PassthroughState.cs)
- [`StartAcquisition`](https://github.com/neurogears/onix-refactor/blob/6dc9e26d38a48b9a0f5c13ffb0a8966236ba2144/OpenEphys.Onix/OpenEphys.Onix/StartAcquisition.cs)

We could potentially internalize some of these as implementation-specific classes, which would simplify both the API and docs effort at this point, but the operators will need to be included.

### PR DESCRIPTION
**Pull request by glopesdev**
Following #175 this PR completes the technical docs by adding missing documentation comments to all classes and members in the public API surface.

Description attributes were also added to all top-level operator classes to allow displaying summary strings in the toolbox.

Fixes #173

### CODE REVIEW

**Review by bparks13 — COMMENTED** [review]
Very nice, thank you @glopesdev!

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureHarpSyncInput.cs [hunk 1]
```diff
@@ -3,22 +3,71 @@
 
 namespace OpenEphys.Onix
 {
+    /// <summary>
+    /// A class for configuring the ONIX breakout board Harp sync input device.
+    /// </summary>
+    /// <remarks>
+    /// Harp is a standard for asynchronous real-time data acquisition and experimental
+    /// control in neuroscience. It includes a clock synchronization protocol which allows
+    /// Harp devices to be connected to a shared clock line and continuously self-synchronise
```
[the remark this answers is not in the document]
Comment (bparks13) [reply OpenEphys.Onix/OpenEphys.Onix/ConfigureHarpSyncInput.cs:12]: "synchronize"

[the remark this answers is not in the document]
Comment (glopesdev) [reply OpenEphys.Onix/OpenEphys.Onix/ConfigureHarpSyncInput.cs:12]: "synchronize"!

File: OpenEphys.Onix/OpenEphys.Onix/ContextTask.cs [hunk 2]
```diff
@@ -9,52 +9,70 @@
...
 
-        private readonly string contextDriver = DefaultDriver;
-        private readonly int contextIndex = DefaultIndex;
+        readonly string contextDriver = DefaultDriver;
+        readonly int contextIndex = DefaultIndex;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ContextTask"/> class.
+        /// </summary>
+        /// <param name="driver"> A string specifying the device driver used to control hardware. </param>
```
[the remark this answers is not in the document]
Comment (glopesdev) [reply OpenEphys.Onix/OpenEphys.Onix/ContextTask.cs:70]: Having a drop-down is a good idea, we can achieve this even without an enum by leveraging [`TypeConverter.GetStandardValues`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.typeconverter.getstandardvalues?view=net-8.0).

**Pull request merged**